stanford.profile

```diff
@@ -276,6 +285,175 @@ function stanford_acsf_tasks() {
+ *
+ * The remaining information is available in the RITM ("Request Item" in ServiceNow) and has been exposed through
+ * an API. Use the sitename as a key and fetch it from the remote api in order
+ * to complete the site installation.
+ */
+function stanford_acsf_tasks_ritm($install_vars) {
+  global $conf;
+
+  // Need this for UI install.
+  require_once DRUPAL_ROOT . '/includes/password.inc';
+
+  // Fetch the json from the ServiceNow (SNOW) API endpoint that exposes the RITM information.
+  $site_name = isset($install_vars['forms']['install_configure_form']['site_name']) ? check_plain($install_vars['forms']['install_configure_form']['site_name']) : NULL;
+
+  if (empty($site_name)) {
+    throw new \Exception("No site_name available. Please pass --site-name to your drush arguments.");
+  }
+
+  // Fetch the information we need from the API.
+  $response = stanford_acsf_tasks_ritm_make_api_request($site_name);
+
+  if (!isset($response['sunetId'])) {
+    throw new \Exception("No sunetId in response body from SNOW");
+  }
+
+  // Pull the primary site owner information out of the response first.
+  $sunet = $response['sunetId'];
+  $name = $response['fullName'];
+  $email = $sunet . "@stanford.edu";
+
+  // Create the primary site owner.
+  $sunetrole = user_role_load_by_name('sso user');
+  $adminrole = user_role_load_by_name('administrator');
+  module_load_include('inc', 'stanford_ssp', 'stanford_ssp.admin');
+
+  if (!is_numeric($sunetrole->rid) || !is_numeric($adminrole->rid)) {
+    throw new \Exception("A role or roles were missing when trying to create a sunet user");
+  }
+
+  // User create payload.
+  $form_state = [
+    'values' => [
+      'sunetid' => $sunet,
+      'name' => $name,
+      'email' => $email,
+      'roles' => [
+        $sunetrole->rid,
+        $adminrole->rid,
+      ],
+    ],
+  ];
+
+  drupal_form_submit('stanford_ssp_add_sso_user', $form_state);
+
+  // Create additional site owners.
+  if (isset($response['webSiteOwners'])) {
+    foreach ($response['webSiteOwners'] as $owner) {
+      // If someone put their own self as an owner or it is a people site,
+      // skip creation of a duplicate account.
+      if ($owner['sunetId'] == $sunet) {
+        continue;
+      }
+
+      // User create payload.
+      $form_state = [
+        'values' => [
+          'sunetid' => $owner['sunetId'],
+          'name' => $owner['fullName'],
+          'email' => $owner['email'],
+          'roles' => [
+            $sunetrole->rid,
+          ],
+        ],
+      ];
+
+      drupal_form_submit('stanford_ssp_add_sso_user', $form_state);
+    }
+  }
+
+  // Set the site title.
+  variable_set('site_name', check_plain($response['webSiteTitle']));
+
+  // Set the site email.
+  variable_set('site_mail', $email);
+}
+
+/**
+ * Fetches JSON information from the ServiceNow (SNOW) API.
+ *
+ * Returns an array that looks something like the following:
+ *  [
+ *    "webSiteTitle": "Meddle",
```

Comment:
I'm going to make a small change here to indicate that `webSiteTitle` can have spaces in it.